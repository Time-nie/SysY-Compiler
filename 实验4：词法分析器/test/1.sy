int a;

int main()
{
    int a;
    a = 0xAF + 021;
    if(a < 5)
        return 1;
    return 0;
}