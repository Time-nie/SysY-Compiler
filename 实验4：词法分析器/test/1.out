	INT	int	1
	IDENTIFIER	a	1	0x55c63f0df410
	SEMICOLON	;	1
	INT	int	3
	IDENTIFIER	main	3	0x55c63f0df8d0
	LPAREN	(	3
	RPAREN	)	3
	LBRACE	{	4
	INT	int	5
	IDENTIFIER	a	5	0x55c63f0df410
	SEMICOLON	;	5
	IDENTIFIER	a	6	0x55c63f0df410
	ASSIGN	=	6
	NUMBER	175	6
	ADD	+	6
	NUMBER	17	6
	SEMICOLON	;	6
	IF	if	7
	LPAREN	(	7
	IDENTIFIER	a	7	0x55c63f0df410
	LESS	<	7
	NUMBER	5	7
	RPAREN	)	7
	RETURN	return	8
	NUMBER	1	8
	SEMICOLON	;	8
	RETURN	return	9
	NUMBER	0	9
	SEMICOLON	;	9
	RBRACE	}	10
